sktime/datatypes/tests/test_convert.py

```diff
@@ -0,0 +1,85 @@
+# -*- coding: utf-8 -*-
+"""Testing machine type converters for scitypes."""
+
+__author__ = ["fkiraly"]
+
+import pytest
+
+from sktime.datatypes._convert import convert, _conversions_defined
+from sktime.datatypes._examples import get_examples
+from sktime.datatypes import SCITYPE_REGISTER
+
+from sktime.utils._testing.deep_equals import deep_equals
+
+SCITYPES = [sci[0] for sci in SCITYPE_REGISTER]
+
+
+@pytest.mark.parametrize("scitype", SCITYPES)
+def test_convert(scitype):
+    """Tests that conversions for scitype agree with from/to example fixtures.
```

Comment:
I think testing logic here is correct, but is hard to follow.  E.g.
* This line `n_fixtures = len(fixtures[mtypes[0]])`only uses the 0 index of `mtypes`. It is not clear upfront that this will not break things, or why this is done.
* In my head, I'd have expected the logic to look something like this:
```
for from_mtype in conv_mat.index:
for to_mtype in conv_mat.columns:
check conversion is defined, if not pass.
loop through examples, do conversions and checks.
```
```suggestion
@pytest.mark.parametrize("scitype", SCITYPES)
def test_convert(scitype):
    """Tests that conversions for scitype agree with from/to example fixtures.
```
```suggestion
@pytest.mark.parametrize("scitype", SCITYPES)
def test_convert(scitype):
    """Tests that conversions for scitype agree with from/to example fixtures.
```

Reply:
>In my head, I'd have expected the logic to look something like this:

Isn't that precisely what happens?

Reply:
there's some additional checking, since some fixtures are "not defined", e.g., if you try to convert a strictly multivariate series to a univariate one - `None` then indicates that it's not possible. The `None` related edge cases need checking.

Reply:
> Isn't that precisely what happens?

Yes, but it is not obvious that is what is happening:
* e.g. the outermost loop ranges through `range(n_fixtures)` whereas I would expect outermost loop to be for columns and indices of conversion matrix
* you loop through rows and indices of conversion matrix indirectly via the variable `mtypes`, whereas I think it would be easier to read if you looped through the indices and columns directly.

Here is something that I think would be clearer:

```
    for from_type in conv_mat.index:
        for to_type in conv_mat.columns:
            for to_fixture, from_fixture in zip(fixtures[to_type], fixtures[from_type])
```

Anyway, for purposes of this review, it is all good as is. If you want to refactor a little bit, you can, otherwise, leave as is.